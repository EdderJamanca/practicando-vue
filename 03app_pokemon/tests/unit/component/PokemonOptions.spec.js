
import { shallowMount } from "@vue/test-utils";
import pokemonOption from "@/page/components/Opciones";

import { pokemons } from "../mocks/pokemons.mock";

describe("PokemonOption Component",()=>{

    let wrapper;

    beforeEach(()=>{
        wrapper=shallowMount(pokemonOption,{
            props:{
                pokemos: pokemons
            }
        })
    });

    test('debe de hacer match con el snapshot',()=>{
        expect(wrapper.html()).toMatchSnapshot();
    })

    test('debe de mostrar las 4 opciones correctamente',()=>{
        const liTags=wrapper.findAll('li');
        // console.log("liTags",liTags)
        expect(liTags.length).toBe(4);

        expect(liTags[0].text()).toBe('pikachu');
        expect(liTags[1].text()).toBe('charmander');
        expect(liTags[2].text()).toBe('venusaur');
        expect(liTags[3].text()).toBe('new');
    })

    test('debe de emitir "selection" con sus respectivos parÃ¡metros al hacer click',()=>{
        const [li1,li2,li3,li4]=wrapper.findAll('li');

        li1.trigger('click')
        li2.trigger('click')
        li3.trigger('click')
        li4.trigger('click')

        expect(wrapper.emitted('selectionPokemon').length).toBe(4);
        expect(wrapper.emitted('selectionPokemon')[0]).toEqual([5]);
        expect(wrapper.emitted('selectionPokemon')[1]).toEqual([10]);
        expect(wrapper.emitted('selectionPokemon')[2]).toEqual([15]);
        expect(wrapper.emitted('selectionPokemon')[3]).toEqual([20]);
    })
})