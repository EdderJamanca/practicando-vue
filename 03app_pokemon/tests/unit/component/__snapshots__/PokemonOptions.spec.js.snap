// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonOption Component debe de hacer match con el snapshot 1`] = `
<div class="options-container">
  <ul>
    <li>pikachu</li>
    <li>charmander</li>
    <li>venusaur</li>
    <li>new</li>
  </ul>
</div>
`;
